dask_ml/model_selection/_split.py

```diff
@@ -502,12 +559,44 @@ def train_test_split(
             blockwise=blockwise,
             random_state=random_state,
         )
-        train_idx, test_idx = next(splitter.split(*arrays))
 
-        train_test_pairs = (
-            (_blockwise_slice(arr, train_idx), _blockwise_slice(arr, test_idx))
-            for arr in arrays
-        )
+        if stratify is not None:
+            # convert to da.Array
+            if not isinstance(stratify, da.Array):
+                stratify = stratify.to_dask_array(lengths=True)
```

Comment:
This will make an additional pass over the data. Ideally we could combine this pass with the pass discovering the frequencies.